gammapy/maps/_sparse.pyx

```diff
@@ -0,0 +1,207 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from __future__ import absolute_import, division, print_function, unicode_literals
+import numpy as np
+cimport numpy as np
+cimport cython
+
+
+@cython.cdivision(True)
+@cython.boundscheck(False)
+def binary_search(np.ndarray[np.int64_t, ndim=1] arr_in, np.int64_t idx,
+                  np.int64_t ilo, np.int64_t ihi):
+    """Perform a binary search for an element ``idx`` in a sorted integer
+    array ``arr_in``.  The search is restricted to the range between
+    ``ilo`` and ``ihi``.  This will return the index of the matching
```

Comment:
Maybe move this sentence describing what the function returns to a `Returns` section, to follow the Numpy docstring standard we use everywhere else?